Sources/StreamChatUI/CommonViews/ContainerStackView.swift

```diff
@@ -336,6 +364,35 @@ public class ContainerStackView: UIView {
     }
 }
 
+// MARK: - UIKit Extension Helpers
+
+extension UIView {
+    func flexible(axis: NSLayoutConstraint.Axis) -> Self {
+        setContentHuggingPriority(.lowest, for: axis)
+        return self
+    }
+}
+
+extension NSLayoutConstraint {
+    func priority(_ p: UILayoutPriority) -> Self {
+        priority = p
+        return self
+    }
+
+    func priority(_ p: Float) -> Self {
+        priority = UILayoutPriority(p)
+        return self
+    }
+}
+
+extension UILayoutPriority {
+    static let lowest = UILayoutPriority(defaultLow.rawValue / 2.0)
+}
+
+extension CGFloat {
+    public static let auto: CGFloat = .infinity
+}
```

Comment:
can we rename this to `systemDefinedContainerSpacing`? `auto` is too general for a public extension on `CGFloat`

Reply:
Actually, I think it is a better idea to create a `ContainerStackView.Spacing`? So that we don't influence the `CGFloat` of the customer. And we can make `ContainerStackView.Spacing` conform to `ExpressibleByFloatLiteral` so that we can still use it like `spacing = 10`.

Example:
```Swift
   public struct Spacing: Equatable, ExpressibleByFloatLiteral {

        public static let auto = Spacing(rawValue: .infinity)

        public init(floatLiteral value: Double) {
            rawValue = CGFloat(value)
        }

        init(rawValue: CGFloat) {
            self.rawValue = rawValue
        }

        var rawValue: CGFloat
    }
```

WDYT?

Reply:
It's a bit overkill but yeah, why not :) However, I think we should expose the raw value publicly then. There's no reason IMO why it shouldn't be.

Reply:
Ah yes it should be public. I made a WIP Implementation here: https://github.com/GetStream/stream-chat-swift/pull/1004/commits/450338621aba6bf23b2134510f1bd5a49b14eb4a, let me know what you think please @VojtaStavik

Reply:
@nuno-vieira all right, let's do it :)